common/sagas/wallet/wallet.ts

```diff
@@ -0,0 +1,250 @@
+    const balances = yield call(getTokenBalances, wallet, tokens);
+
+    // Save the non-zero tokens, tokens previously tracked, and custom tokens
+    const customTokens = yield select(getCustomTokens);
+    const oldConfig = yield call(loadWalletConfig, wallet);
+    const nonEmptyTokens = Object.keys(balances).filter(symbol => {
+      if (balances[symbol] && !balances[symbol].balance.isZero()) {
+        return true;
+      }
+      if (oldConfig.tokens && oldConfig.tokens.includes(symbol)) {
+        return true;
+      }
+      if (customTokens.find(token => token.symbol === symbol)) {
+        return true;
+      }
+    });
+    const config = yield call(saveWalletConfig, wallet, { tokens: nonEmptyTokens });
+    yield put(setWalletConfig(config));
+
+    // Finish off setTokenBalances, sending only:
+    // 1. Tokens with value
+    // 2. Tokens that were tracked previously
+    // 3. Custom tokens
+    yield put(setTokenBalancesFulfilled(balances));
+  } catch (err) {
+    console.error('Failed to scan for tokens', err);
+    yield put(setTokenBalancesRejected());
+  }
+}
+
+export function* handleSetWalletTokens(action: SetWalletTokensAction): SagaIterator {
+  const wallet: null | IWallet = yield select(getWalletInst);
+  if (!wallet) {
+    return;
+  }
+
+  const config = yield call(saveWalletConfig, wallet, { tokens: action.payload });
+  yield put(setWalletConfig(config));
+}
+
+export function* updateBalances(): SagaIterator {
+  yield fork(updateAccountBalance);
+  yield fork(updateTokenBalances);
+}
+
+export function* handleNewWallet(): SagaIterator {
+  yield call(updateWalletConfig);
+  yield fork(updateBalances);
+}
+
+export function* updateWalletConfig(): SagaIterator {
+  const wallet: null | IWallet = yield select(getWalletInst);
+  if (!wallet) {
+    return;
+  }
+  const config = yield call(loadWalletConfig, wallet);
+  yield put(setWalletConfig(config));
+}
+
+export function* unlockPrivateKey(action: UnlockPrivateKeyAction): SagaIterator {
+  let wallet: IWallet | null = null;
+  const { key, password } = action.payload;
+
+  try {
+    wallet = getPrivKeyWallet(key, password);
+  } catch (e) {
+    yield put(showNotification('danger', translate('INVALID_PKEY')));
+    return;
+  }
+  yield put(setWallet(wallet));
+}
+
+export function* unlockKeystore(action: UnlockKeystoreAction): SagaIterator {
+  const { file, password } = action.payload;
+  let wallet: null | IWallet = null;
+
+  try {
+    wallet = getKeystoreWallet(file, password);
+  } catch (e) {
+    yield put(showNotification('danger', translate('ERROR_6')));
+    return;
+  }
+
+  // TODO: provide a more descriptive error than the two 'ERROR_6' (invalid pass) messages above
+  yield put(setWallet(wallet));
+}
+
+export function* unlockMnemonic(action: UnlockMnemonicAction): SagaIterator {
+  let wallet;
+  const { phrase, pass, path, address } = action.payload;
+
+  try {
+    wallet = MnemonicWallet(phrase, pass, path, address);
```

Comment:
This should be the result of a `yield call`

Reply:
90% of this file is just moving from `sagas/wallet.ts` to `sagas/wallet/wallet.ts`. Unfortunately git didn't seem to realize that, but this (And a lot of the other typing comments) are from past changes. I can try to go through and clean them up though.

Reply:
Ah noted, sorry about that